modules/contacts/hm-contacts.php

```diff
@@ -142,13 +142,66 @@ public function reset() {
         $this->data = array();
     }
 
-    public function page($page, $size) {
+    public function page($page, $size, $data = null) {
         if ($page < 1) {
             return array();
         }
-        return array_slice($this->data, (($page - 1)*$size), $size, true);
+        if ($data === null) {
+            $data = $this->data;
+        }
+        return array_slice($data, (($page - 1)*$size), $size, true);
     }
 
+    public function group_by($column = 'group') {
```

Comment:
Can you refactor the code to group by a string version of the column name? I don't see a reason we group by a callable or support cases for array or object contacts. You can refactor calling code to ensure we are working with a standard set of data - either array of arrays or array of contact objects.